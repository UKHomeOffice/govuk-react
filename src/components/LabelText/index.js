// https://github.com/alphagov/govuk-frontend/tree/master/src/components/

import glamorous from "glamorous";
import { BLACK } from "govuk-colours";
import { MEDIA_QUERIES } from "../../constants/index";

const LabelText = glamorous.span(
  {
    fontFamily: '"nta", Arial, sans-serif',
    WebkitFontSmoothing: "antialiased",
    MozOsxFontSmoothing: "grayscale",
    display: "block",
    clear: "none",
    fontWeight: 400,
    fontSize: "16px",
    lineHeight: "1.25",
    color: `${BLACK}`,
    paddingBottom: "2px",
    [MEDIA_QUERIES.LARGESCREEN]: {
      fontSize: "19px",
      lineHeight: "1.31579"
    }
  },
  ({ error }) => ({
    fontWeight: error ? 700 : LabelText.fontWeight
  })
);

export default LabelText;
