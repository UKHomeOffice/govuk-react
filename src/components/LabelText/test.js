import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";

import LabelText from "./";

describe("LabelText", () => {
  const example = "example";
  const wrapper = <LabelText>{example}</LabelText>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
  });
});
