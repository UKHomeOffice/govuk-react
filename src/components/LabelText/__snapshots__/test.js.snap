// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelText matches snapshot: enzyme.mount 1`] = `
.glamor-0,
[data-glamor-0] {
  font-family: "nta", Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: block;
  clear: none;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  padding-bottom: 2px;
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    font-size: 19px;
    line-height: 1.31579;
  }
}

<glamorous(span)>
  <span
    className="glamor-0"
  >
    example
  </span>
</glamorous(span)>
`;
